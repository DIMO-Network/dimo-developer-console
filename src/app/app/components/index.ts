export * from './OnboardingCard';
export * from './Onboarding';
