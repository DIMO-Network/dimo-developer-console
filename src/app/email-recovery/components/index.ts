export * from './EmailRecoveryForm';
export * from './CheckEmail';