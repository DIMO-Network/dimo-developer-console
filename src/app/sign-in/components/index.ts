export * from './OtpInputForm';
export * from './PasskeyLogin';
export * from './SignInMethodForm';
