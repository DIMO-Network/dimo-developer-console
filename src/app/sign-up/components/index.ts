export * from './BuildForForm';
export * from './CompanyInfoForm';
export * from './SignUpWith';
export * from './UserInfoForm';
