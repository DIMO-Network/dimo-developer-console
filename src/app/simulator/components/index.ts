export { View } from './View';
export { Header } from './Header';
export { SimulatedVehiclesSection } from './SimulatedVehiclesSection';
export { CreateSimulatedVehicleModal } from './CreateSimulatedVehicleModal';
