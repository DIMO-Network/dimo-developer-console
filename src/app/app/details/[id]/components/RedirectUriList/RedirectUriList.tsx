import { type FC } from 'react';
import { TrashIcon } from '@heroicons/react/24/outline';

import { IApp } from '@/types/app';
import { Table } from '@/components/Table';
import { Toggle } from '@/components/Toggle';

interface IProps {
  app: IApp;
}

export const RedirectUriList: FC<IProps> = ({ app }) => {
  const renderToggleStatus = (data: Record<string, string>) => {
    const value = data['status'];
    return <Toggle checked={Boolean(value)} />;
  };

  const renderDeleteRedirectUriAction = () => {
    return <TrashIcon className="w-5 h-5" />;
  };
  return (
    <Table
      columns={[{ name: 'redirectUri' }]}
      data={app.redirectUris}
      actions={[renderToggleStatus, renderDeleteRedirectUriAction]}
    />
  );
};

export default RedirectUriList;
