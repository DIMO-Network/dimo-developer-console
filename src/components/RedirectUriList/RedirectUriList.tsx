import { get } from 'lodash';
import * as Sentry from '@sentry/nextjs';
import React, { useState, type FC, useContext } from 'react';
import { TrashIcon } from '@heroicons/react/24/outline';
import { Table } from '@/components/Table';
import { useSetRedirectUri } from '@/hooks';
import { Button } from '@/components/Button';
import { DeleteConfirmationModal } from '../DeleteConfirmationModal';
import { LoadingStatusContext } from '@/context/LoadingStatusContext';
import { withLoadingStatus } from '@/hoc';
import { CopyButton } from '@/components/CopyButton';

import '@/components/Button/Button.css';

interface RedirectUri {
  uri: string;
}

interface IProps {
  redirectUris: RedirectUri[] | undefined;
  refreshData: () => void;
  tokenId: number;
  isOwner: boolean;
}

const RedirectUriListComponent: FC<IProps> = ({
  redirectUris = [],
  refreshData,
  tokenId,
  isOwner,
}) => {
  const { setLoadingStatus } = useContext(LoadingStatusContext);
  const [uriToDelete, setUriToDelete] = useState<string>();

  const setRedirectUri = useSetRedirectUri(tokenId);

  const handleError = (error: unknown) => {
    Sentry.captureException(error);
    const code = get(error, 'code', null);
    if (code === 4001)
      setLoadingStatus({
        label: 'The transaction was denied',
        status: 'error',
      });
    else setLoadingStatus({ label: 'Something went wrong', status: 'error' });
  };

  const handleDelete = async (uri: string) => {
    try {
      setLoadingStatus({
        label: 'Deleting the selected redirect URI',
        status: 'loading',
      });
      await setRedirectUri(uri, false);
      setLoadingStatus({ label: 'Redirect URI deleted', status: 'success' });
      refreshData();
    } catch (error: unknown) {
      handleError(error);
    }
  };

  const onConfirmDelete = async () => {
    if (!uriToDelete) {
      throw new Error('No uri to delete');
    }
    await handleDelete(uriToDelete);
    setUriToDelete(undefined);
  };

  const renderCopyRedirectUriAction = ({ uri }: RedirectUri, index: number) => {
    return (
      <CopyButton
        key={`copy-action-${index}`}
        value={uri}
        onCopySuccessMessage={'Redirect URI copied!'}
        className={'button table-action-button'}
      />
    );
  };

  const renderDeleteRedirectUriAction = ({ uri }: RedirectUri, index: number) => {
    return (
      isOwner && (
        <Button
          className={'table-action-button'}
          title="Delete redirect URI"
          type="button"
          onClick={() => setUriToDelete(uri)}
          key={`delete-action-${index}`}
        >
          <TrashIcon className="w-5 h-5 cursor-pointer" />
        </Button>
      )
    );
  };

  return (
    <>
      <Table
        columns={[{ name: 'uri', label: 'Authorized URIs' }]}
        data={redirectUris}
        actions={[renderCopyRedirectUriAction, renderDeleteRedirectUriAction]}
      />
      <DeleteConfirmationModal
        isOpen={!!uriToDelete}
        title={'Are you sure you want to delete this Redirect URI?'}
        subtitle={'You will no longer be able to use this redirect URI in your app.'}
        onConfirm={onConfirmDelete}
        onCancel={() => {
          setUriToDelete(undefined);
        }}
        confirmButtonClassName={'error'}
      />
    </>
  );
};

export const RedirectUriList = withLoadingStatus(RedirectUriListComponent);
