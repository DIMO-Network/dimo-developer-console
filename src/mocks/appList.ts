import { IApp } from '@/types/app';

export const appListMock: IApp[] = [
  {
    name: 'Test App',
    scope: 'sandbox',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Fleet Management App',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Data Analytics',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
];
