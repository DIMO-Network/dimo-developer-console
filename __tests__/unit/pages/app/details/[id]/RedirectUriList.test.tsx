import { fireEvent, render, screen } from '@testing-library/react';

import { RedirectUriList } from '@/app/app/details/[id]/components/RedirectUriList';
import { appListMock } from '@/mocks/appList';

const [app] = appListMock;

describe('RedirectUriList', () => {
  it('renders the redirect uri list', () => {
    const { container } = render(<RedirectUriList app={app} />);

    const [redirectUri] = app.redirectUris;
    const [redirectUriElm] = screen.getAllByText(redirectUri.redirectUri);
    const [toggleElm] = screen.getAllByLabelText('toggle');

    fireEvent.click(toggleElm);

    expect(redirectUriElm).toBeInTheDocument();
    expect(toggleElm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
