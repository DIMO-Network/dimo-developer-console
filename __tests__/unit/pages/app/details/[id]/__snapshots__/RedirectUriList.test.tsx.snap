// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedirectUriList renders the redirect uri list 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          class="custom-table-column"
          scope="col"
        >
          redirectUri
        </th>
        <th
          class="table-action-column"
          scope="col"
        >
          <span
            class="sr-only"
          >
            Actions
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="table-body"
    >
      <tr>
        <td
          class="table-cell"
        >
          https://google.com/
        </td>
        <td
          class="table-action-cell"
        >
          <button
            aria-checked="false"
            aria-label="toggle"
            class="custom-toggle group"
            data-changing=""
            data-headlessui-state="changing"
            id="headlessui-switch-:r0:"
            role="switch"
            tabindex="0"
            type="button"
          >
            <span
              class="sr-only"
            >
              Use setting
            </span>
            <span
              aria-hidden="true"
              class="base"
            />
            <span
              aria-hidden="true"
              class="inactive bar"
            />
            <span
              aria-hidden="true"
              class="inactive dot"
            />
          </button>
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          https://google.com.co/
        </td>
        <td
          class="table-action-cell"
        >
          <button
            aria-checked="true"
            aria-label="toggle"
            class="custom-toggle group"
            data-checked=""
            data-headlessui-state="checked"
            id="headlessui-switch-:r1:"
            role="switch"
            tabindex="0"
            type="button"
          >
            <span
              class="sr-only"
            >
              Use setting
            </span>
            <span
              aria-hidden="true"
              class="base"
            />
            <span
              aria-hidden="true"
              class="active bar"
            />
            <span
              aria-hidden="true"
              class="active dot"
            />
          </button>
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
